test/asyncio_tests/test_asyncio_change_stream.py

```diff
@@ -260,3 +263,24 @@ async def test_database(self):
             i += 1
             if i == 3:
                 break
+
+    @asyncio_test(timeout=10)
+    async def test_iterate_more_streams_than_workers(self):
+        # Create more tasks running ChangeStream.next than there are worker
+        # threads, and then ensure that other tasks can still run.
+        streams = [self.collection.watch() for _ in range(max_workers)]
+        tasks = [stream.next() for stream in streams]
```

Comment:
How come this doesn't end up raising StopAsyncIteration() ? IIUC, `asyncio.gather` should propagate that error when it occurs...

Reply:
Notice the intentionally missing call to `await`. We're calling stream.next() to create a task but not actually awaiting it until the call to gather().

Reply:
Right. But `gather` should `await` each of these and propagate the StopAsyncInteration exceptions raised therein because we are not setting `return_exceptions=True`. As per docs of `gather`: 

> If return_exceptions is False (default), the first raised exception is immediately propagated to the task that awaits on gather(). Other awaitables in the aws sequence won’t be cancelled and will continue to run.

Reply:
The next() call returns a document because we insert documents to the same collection.